src/backend/marsha/core/serializers/live_session.py

```diff
@@ -143,34 +142,61 @@ def validate(self, attrs):
                         }
                     )
                 # Make sure we have the anonymous_id
-                if not attrs.get("anonymous_id"):
+                if not validated_data.get("anonymous_id"):
                     raise serializers.ValidationError(
                         {"anonymous_id": "Anonymous id is mandatory."}
                     )
 
                 # Control this email hasn't already been used for this video in the public case
                 if LiveSession.objects.filter(
                     consumer_site=None,
-                    email=attrs["email"],
+                    email=validated_data["email"],
                     lti_id=None,
                     video=video,
                 ).exists():
                     raise serializers.ValidationError(
                         {
-                            "email": f"{attrs['email']} is already registered "
+                            "email": f"{validated_data['email']} is already registered "
                             "for this video, consumer site and course."
                         }
                     )
 
-                attrs["consumer_site"] = None
-                attrs["lti_id"] = None
-
-        return super().validate(attrs)
+                validated_data["consumer_site"] = None
+                validated_data["lti_id"] = None
 
-    def create(self, validated_data):
         validated_data["is_registered"] = True
         return super().create(validated_data)
 
+    def update(self, instance, validated_data):
+        """Do not allow to update some fields."""
+        # User here is a video as it comes from the JWT
+        # It is named "user" by convention in the `rest_framework_simplejwt` dependency we use.
+        user = self.context["request"].user
+        updatable_fields = ["is_registered", "email"]
```

Comment:
```suggestion
        updatable_fields = ["is_registered", "email"]
```
shouldn't we use a specific serializer in the API for the update directly targeting the only field that can be udpaded?

Reply:
I don't think so. Using the same allow us to return the complete `live_session` record in the response.